Separate tracker and genesis jobs from vrfy job

KateFriedman-NOAA opened the pull request:
# Description

This PR creates new individual jobs for the tracker and genesis jobs that are being pulled out of the `vrfy` job (issue #235). The `tracker` and `genesis` jobs will be on by default while the `genesis_fsu` job will be off by default. This mimics the current settings in the `vrfy` job. 

No enhancements or upgrades are made to the jobs themselves in this PR, only to separate them from the `vrfy` job. However, one change compared to how they currently run in the `vrfy` job is to no longer run the `tracker` in the gdas suite. This mimics how the `ens_tracker` package is run in operations (gfs only).

Changes in this PR:

1. Create rocoto job scripts for tracker/genesis jobs.
2. Create configs for tracker/genesis jobs, as well as `config.tropcy` for shared settings.
3. Add tracker/genesis jobs to `config.resources`; use resources from the ecf scripts for `ens_tracker` jobs in operations
4. Add tracker/genesis jobs to rocoto setup for both cycled and forecast-only modes.
5. Update arguments for `jjob_header.sh` script in tracker/genesis JJOB scripts to use specific job names and configs instead of vrfy.
6. Rename `tracker_ver` to `ens_tracker_ver` in version files to mimic variable within tracker package.
7. Rename and move tracker/genesis switches to `config.base`.
8. Remove tracker/genesis jobs from `vrfy` job (`jobs/rocoto/vrfy` and `config.vrfy`).
9. Reduce vrfy job resources to accommodate tasks having been removed.

Refs #235
Resolves #1988
Resolves #2048

# Type of change

- New feature (adds functionality)

# Change characteristics
- Is this a breaking change (a change in existing functionality)? NO
- Does this change require a documentation update? NO

# How has this been tested?

- Cycled atmos-only testing on WCOSS2 (test covered period with Typhoon Rai in western Pacific)
- Forecast-only coupled testing on WCOSS2 (no storms in test date, just tested setup)

# Checklist
- [x] Any dependent changes have been merged and published
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] New and existing tests pass with my changes
- [x] I have made corresponding changes to the documentation if necessary

KateFriedman-NOAA committed:
Create rocoto job scripts for tracker/genesis

Refs #1988

KateFriedman-NOAA committed:
Add tracker/genesis settings to config.base

- Add flags to turn on/off the tracker and genesis jobs
- Add HOMEens_tracker path setting
- Settings previously found on config.vrfy

Refs #1988

KateFriedman-NOAA committed:
Remove tracker/genesis settings from config.vrfy

Refs #1988

KateFriedman-NOAA committed:
Remove tracker/genesis from vrfy rocoto job script

Refs #1988

KateFriedman-NOAA committed:
Create configs for tracker/genesis jobs

Refs #1988

KateFriedman-NOAA committed:
Add tracker/genesis to resource config

Refs #1988

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/tracker_jobs_separate

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/tracker_jobs_separate

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/tracker_jobs_separate

KateFriedman-NOAA committed:
Add tracker and genesis jobs to setup

When jobs turned on:
- Add jobs to rocoto mesh.
- Add configs to list of configs to source.
- Add jobs to gfsarch dependency list.

Refs #1988

KateFriedman-NOAA committed:
Reduce vrfy job resources

- Drop walltime to 10 mins.
- Drop node usage to 1.
- Remove exclusive flag, no longer needed.

The job needs very little resources now ahead of the final
pieces being removed.

Refs #1988

KateFriedman-NOAA committed:
Update jjob_header arguments for tracker/genesis

Refs #1988

KateFriedman-NOAA committed:
Change tracker_ver to ens_tracker_ver

Change version variable to match what JJOB scripts use

Refs #1988

KateFriedman-NOAA committed:
Create and use config.tropcy

- Create new config for common tracker and genesis settings.
- Update tracker and genesis configs to source config.tropcy.

Refs #1988

KateFriedman-NOAA committed:
Tropcy config updates

- Move HOMEens_tracker setting to config.tropcy.
- Delete COMINsyn setting from config.tropcy (will use
setting from config.base).

Refs #1988

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/tracker_jobs_separate

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/tracker_jobs_separate

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/tracker_jobs_separate

KateFriedman-NOAA committed:
Remove extra line in config.resources

Refs #1988

KateFriedman-NOAA committed:
Add tracker/genesis to forecast-only mode

Also correct syntax for adding jobs to tasks.py

Refs #1988

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/tracker_jobs_separate

KateFriedman-NOAA committed:
Merge remote-tracking branch 'upstream/develop' into feature/tracker_jobs_separate

* upstream/develop:
  Refine fit2obs job in rocoto mesh (#2047)

aerorahul reviewed (approved):
looks good.
`vrfy` job is significantly trimmed down.

KateFriedman-NOAA committed:
Update post dependencies for archive job

- Adjust post metatask dependency to always be added, instead of
only when there aren't other dependencies.
- Add postanl job to archive dependencies.

Refs #2048

KateFriedman-NOAA commented:
@aerorahul @WalterKolczynski-NOAA I have folded in changes for issue #2048 into the branch for this PR.

See setup script changes https://github.com/KateFriedman-NOAA/global-workflow/commit/07f97a845973936ca7515a19455f70f5658b70e7 and how the archive dependencies will now look in the xml: https://github.com/NOAA-EMC/global-workflow/issues/2048#issuecomment-1811049321

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/tracker_jobs_separate

WalterKolczynski-NOAA commented:
Looks right to me.
